### PR TITLE
Some Catalyst API annotations and corrections

### PR DESCRIPTION
**Pull request by chamons**
- Part of https://github.com/xamarin/xamarin-macios/issues/10216
- This includes most of the smaller SDKs.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

**Review by spouliot — APPROVED** [review]
Intro might (should) complain about the `[MacCatalyst (13,0)]`
since it's below what is defined in `src/MinimumVersions.cs`

File: src/callkit.cs [hunk 1]
```diff
@@ -296,8 +296,7 @@ interface CXCallDirectoryManager {
 		[Export ("getEnabledStatusForExtensionWithIdentifier:completionHandler:")]
 		void GetEnabledStatusForExtension (string identifier, Action<CXCallDirectoryEnabledStatus, NSError> completion);
 
-		[NoMacCatalyst]
-		[NoWatch, NoTV, NoMac, iOS (13,4)]
+		[NoWatch, NoTV, NoMac, iOS (13,4), MacCatalyst (10,14)]
```
Comment (rolfbjarne) [inline src/callkit.cs:299]: [MacCatalyst (10, 14)] is wrong.

**Review by rolfbjarne — APPROVED** [review]

**Comment by chamons** [comment]
@spouliot 

```
 git grep MacCatalyst | grep 13 | wc -l
     160
```

Should I fix the ones in this PR?

**Comment by spouliot** [comment]
@chamons `13,1` is _allowed_ by the test - since `<=` resulted in too many things to fix.

and `13,2` and later must be allowed.

**Comment by spouliot** [comment]
https://github.com/xamarin/xamarin-macios/blob/209486aa5a0cf7df2ce17f00d2b1a2376495004c/tests/introspection/ApiAvailabilityTest.cs#L232

**Comment by chamons** [comment]
git grep MacCatalyst | grep -E "13, 0|13,0" | wc -l
     105

Still a bunch to fix already landed today (xkit.cs and other files I didn't touch).

Should I fix the ones in this PR? Bump them to 13,1?

**Review by mandel-macaque — APPROVED** [review]

**Comment by chamons** [comment]
🤔 The failure on the non-expected tests:

```
App could not be built .
```

**Comment by rolfbjarne** [comment]
Looks like while the headers say the Chip framework is available for Mac Catalyst, there's no way to link with it:

```
ld: building for Mac Catalyst, but linking in .tbd built for macOS, file '/Applications/Xcode_13.0.0-beta5.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/System/Library/Frameworks//CHIP.framework/CHIP.tbd' for architecture x86_64
```

Maybe try to use the Chip framework from Xcode, but I suspect the fix is to remove the framework from our Mac Catalyst bindings.

**Review by tj-devel709 — APPROVED** [review]

**Comment by mandel-macaque** [comment]
@chamons  the fix is to remove CHIP, which is why I did it in the PR that added it. CHIP is not present in catalyst.

**Comment by chamons** [comment]
Ok, I finally got a clean intro/xtro locally, so this spam should hopefully be over.

**Review by dalexsoto — APPROVED** [review]

File: tests/xtro-sharpie/MacCatalyst-IntentsUI.ignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# fails on intro rdar: https://github.com/xamarin/maccore/issues/2516
+!missing-selector! +INImage::imageSizeForIntentResponse: not bound
+!missing-selector! +INImage::imageWithCGImage: not bound
+!missing-selector! +INImage::imageWithUIImage: not bound
+!missing-selector! INImage::fetchUIImageWithCompletion: not bound
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/MacCatalyst-IntentsUI.ignore:5]: Same!

File: tests/xtro-sharpie/MacCatalyst-CHIP.ignore [hunk 3]
```diff
@@ -398,3 +398,153 @@
...
+!missing-type! CHIPKeypadInput not bound
+!missing-type! CHIPMediaInput not bound
+!missing-type! CHIPMediaPlayback not bound
+!missing-type! CHIPRelativeHumidityMeasurement not bound
+!missing-type! CHIPSoftwareDiagnostics not bound
+!missing-type! CHIPTargetNavigator not bound
+!missing-type! CHIPTrustedRootCertificates not bound
+!missing-type! CHIPTvChannel not bound
+!missing-type! CHIPWakeOnLan not bound
+!missing-type! CHIPWindowCovering not bound
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/MacCatalyst-CHIP.ignore:550]: No big deal, but missing eol :)

**Review by rolfbjarne — APPROVED** [review]

**Comment by chamons** [comment]
Test issues are fixed in https://github.com/xamarin/xamarin-macios/pull/12776

**Comment by chamons** [comment]
One part not reading xtro results locally beyond sanity verification and one part Apple marking base classes without marking categories. 

This ~time~ commit with feeling.

**Comment by chamons** [comment]
Failures:

```
15:03:17.5378800 2021-09-20 15:03:17.537 monotouchtest[20909:37529800] 	[FAIL] InvokeTest


15:03:49.4599670 2021-09-20 15:03:49.459 monotouchtest[20909:37529800] 	[FAIL] Bug17793 :   Not signalled twice in 5s
15:03:49.4601200   Expected: True
15:03:49.4601480   But was:  False

15:04:01.6310260 2021-09-20 15:04:01.630 monotouchtest[20909:37529800] 	[FAIL] Bug2443 :   Not signalled twice in 5s
15:04:01.6310810   Expected: True
15:04:01.6310930   But was:  False
```

**Comment by chamons** [comment]
Invoke Test - https://github.com/xamarin/maccore/issues/2451
Timer Test (at least one, same message) - https://github.com/xamarin/maccore/issues/2443

**Pull request merged**